src/code_80005FD0.c

```diff
@@ -1406,8 +1406,7 @@ void func_800090F0(s32 playerId, Player *player) {
 f32 func_80009258(UNUSED s32 playerId, f32 arg1, f32 arg2) {
     f32 temp_f2 = D_8016344C - arg2;
     f32 temp_f12 = arg1 - D_8016344C;
-
-    return gCourseTimer - ( (0.01666666f * temp_f2) / (temp_f2 + temp_f12) );
+    return gCourseTimer - ( (COURSE_TIMER_ITER * temp_f2) / (temp_f2 + temp_f12) );
```

Comment:
This might be the line that makes it not match.

It's possible the f at the end is required. Or num 6's are diff